packages/esm-ward-app/src/ward-view/ward-view.component.tsx

```diff
@@ -61,10 +61,16 @@ const WardViewWithBedManagement = () => {
       const wardPatients: WardPatient[] = patients.map((patient) => {
         const inpatientAdmission = inpatientAdmissionsByPatientUuid.get(patient.uuid);
         if (inpatientAdmission) {
-          return { patient: inpatientAdmission.patient, visit: inpatientAdmission.visit, admitted: true };
+          return { ...inpatientAdmission, admitted: true };
         } else {
           // for some reason this patient is in a bed but not in the list of admitted patients, so we need to use the patient data from the bed endpoint
-          return { patient: patient, visit: null, admitted: false };
+          return {
+            patient: patient,
+            visit: null,
+            admitted: false,
+            encounterAssigningToCurrentInpatientLocation: null,
+            firstAdmissionOrTransferEncounter: null, // TODO: populate this after O3-3704
```

Comment:
FWIW, we still won't be able to populate this after O3-3704... this is the case when for some reason the patient hasn't been admitted and only the bed management endpoint is returning the patient.

Reply:
firstAdmissionOrTransferEncounter will be null if the patient is not admitted.  There will be no TODO in this case, and the "timeSinceAdmission" should presumably be null and hidden (since they aren't admitted).

However, the EncounterAssigningToCurrentInpatientLocation could (and probably should) be taken from the encounter associated with the patient bed assignment that first put them on the ward, and then this could still drive the "Time on Ward" element accurately.

Reply:
filed https://openmrs.atlassian.net/browse/BED-13